### ISSUE TITLE
Failed to covert StoreFragments type

### ISSUE THREAD

**Issue by vachillo** [paulcwarren/spring-content#252]
Running a spring boot app with a simple spring content store, and trying to expose it via spring content rest:

the store:
```
@StoreRestResource(path = "documents", linkRel = "documents")
public interface DocumentStore extends ContentStore<Document, String>, Renderable<Document> {

}
```
the app boots fine but no store is exposed at `localhost:8080/`. I know spring data rest is working fine because i can see the other spring data repository that i'm exposing.

when running in debug, during startup i see:
```
org.springframework.core.convert.ConversionFailedException: Failed to convert from type [internal.org.springframework.content.commons.config.StoreFragments] to type [@org.springframework.beans.factory.annotation.Autowired internal.org.springframework.content.commons.config.StoreFragments] for value '[internal.org.springframework.content.commons.config.StoreFragment@1fc1c7e]'; nested exception is java.lang.IllegalArgumentException: Could not instantiate Collection type: internal.org.springframework.content.commons.config.StoreFragments
	at org.springframework.core.convert.support.ConversionUtils.invokeConverter(ConversionUtils.java:47) ~[spring-core-5.2.6.RELEASE.jar:5.2.6.RELEASE]
	at org.springframework.core.convert.support.GenericConversionService.convert(GenericConversionService.java:191) ~[spring-core-5.2.6.RELEASE.jar:5.2.6.RELEASE]
	at org.springframework.beans.TypeConverterDelegate.convertIfNecessary(TypeConverterDelegate.java:129) ~[spring-beans-5.2.6.RELEASE.jar:5.2.6.RELEASE]
	at org.springframework.beans.AbstractNestablePropertyAccessor.convertIfNecessary(AbstractNestablePropertyAccessor.java:585) ~[spring-beans-5.2.6.RELEASE.jar:5.2.6.RELEASE]
	at org.springframework.beans.AbstractNestablePropertyAccessor.convertForProperty(AbstractNestablePropertyAccessor.java:604) ~[spring-beans-5.2.6.RELEASE.jar:5.2.6.RELEASE]
	at org.springframework.beans.BeanWrapperImpl.convertForProperty(BeanWrapperImpl.java:219) ~[spring-beans-5.2.6.RELEASE.jar:5.2.6.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.convertForProperty(AbstractAutowireCapableBeanFactory.java:1748) ~[spring-beans-5.2.6.RELEASE.jar:5.2.6.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.applyPropertyValues(AbstractAutowireCapableBeanFactory.java:1704) ~[spring-beans-5.2.6.RELEASE.jar:5.2.6.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.populateBean(AbstractAutowireCapableBeanFactory.java:1444) ~[spring-beans-5.2.6.RELEASE.jar:5.2.6.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:594) ~[spring-beans-5.2.6.RELEASE.jar:5.2.6.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:517) ~[spring-beans-5.2.6.RELEASE.jar:5.2.6.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:323) ~[spring-beans-5.2.6.RELEASE.jar:5.2.6.RELEASE]
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:226) ~[spring-beans-5.2.6.RELEASE.jar:5.2.6.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:321) ~[spring-beans-5.2.6.RELEASE.jar:5.2.6.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:202) ~[spring-beans-5.2.6.RELEASE.jar:5.2.6.RELEASE]
	at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:276) ~[spring-beans-5.2.6.RELEASE.jar:5.2.6.RELEASE]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1306) ~[spring-beans-5.2.6.RELEASE.jar:5.2.6.RELEASE]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1226) ~[spring-beans-5.2.6.RELEASE.jar:5.2.6.RELEASE]
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.inject(AutowiredAnnotationBeanPostProcessor.java:640) ~[spring-beans-5.2.6.RELEASE.jar:5.2.6.RELEASE]
	at org.springframework.beans.factory.annotation.InjectionMetadata.inject(InjectionMetadata.java:130) ~[spring-beans-5.2.6.RELEASE.jar:5.2.6.RELEASE]
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.postProcessProperties(AutowiredAnnotationBeanPostProcessor.java:399) ~[spring-beans-5.2.6.RELEASE.jar:5.2.6.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.populateBean(AbstractAutowireCapableBeanFactory.java:1422) ~[spring-beans-5.2.6.RELEASE.jar:5.2.6.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:594) ~[spring-beans-5.2.6.RELEASE.jar:5.2.6.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:517) ~[spring-beans-5.2.6.RELEASE.jar:5.2.6.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:323) ~[spring-beans-5.2.6.RELEASE.jar:5.2.6.RELEASE]
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:226) ~[spring-beans-5.2.6.RELEASE.jar:5.2.6.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:321) ~[spring-beans-5.2.6.RELEASE.jar:5.2.6.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:202) ~[spring-beans-5.2.6.RELEASE.jar:5.2.6.RELEASE]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:895) ~[spring-beans-5.2.6.RELEASE.jar:5.2.6.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:878) ~[spring-context-5.2.6.RELEASE.jar:5.2.6.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:550) ~[spring-context-5.2.6.RELEASE.jar:5.2.6.RELEASE]
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:143) ~[spring-boot-2.3.0.RELEASE.jar:2.3.0.RELEASE]
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:758) ~[spring-boot-2.3.0.RELEASE.jar:2.3.0.RELEASE]
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:750) ~[spring-boot-2.3.0.RELEASE.jar:2.3.0.RELEASE]
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:397) ~[spring-boot-2.3.0.RELEASE.jar:2.3.0.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:315) ~[spring-boot-2.3.0.RELEASE.jar:2.3.0.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1237) ~[spring-boot-2.3.0.RELEASE.jar:2.3.0.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1226) ~[spring-boot-2.3.0.RELEASE.jar:2.3.0.RELEASE]
	at com.optum.apollo.orbit.api.storage.OrbitStorageApiApplication.main(OrbitStorageApiApplication.java:21) ~[classes/:na]
Caused by: java.lang.IllegalArgumentException: Could not instantiate Collection type: internal.org.springframework.content.commons.config.StoreFragments
	at org.springframework.core.CollectionFactory.createCollection(CollectionFactory.java:212) ~[spring-core-5.2.6.RELEASE.jar:5.2.6.RELEASE]
	at org.springframework.core.convert.support.CollectionToCollectionConverter.convert(CollectionToCollectionConverter.java:81) ~[spring-core-5.2.6.RELEASE.jar:5.2.6.RELEASE]
	at org.springframework.core.convert.support.ConversionUtils.invokeConverter(ConversionUtils.java:41) ~[spring-core-5.2.6.RELEASE.jar:5.2.6.RELEASE]
	... 38 common frames omitted
Caused by: java.lang.NoSuchMethodException: internal.org.springframework.content.commons.config.StoreFragments.<init>()
	at java.base/java.lang.Class.getConstructor0(Class.java:3350) ~[na:na]
	at java.base/java.lang.Class.getDeclaredConstructor(Class.java:2554) ~[na:na]
	at org.springframework.util.ReflectionUtils.accessibleConstructor(ReflectionUtils.java:185) ~[spring-core-5.2.6.RELEASE.jar:5.2.6.RELEASE]
	at org.springframework.core.CollectionFactory.createCollection(CollectionFactory.java:208) ~[spring-core-5.2.6.RELEASE.jar:5.2.6.RELEASE]
	... 40 common frames omitted
```
and when trying to run test method calls on the content store programmatically after boot, nothing happens. Intellij is also telling me it cant find an autowire candidate for DocumentStore.

my pom: 
```
<properties>
        <java.version>11</java.version>
        <maven.compiler.target>${java.version}</maven.compiler.target>
        <maven.compiler.source>${java.version}</maven.compiler.source>

        <spring.boot.version>2.3.0.RELEASE</spring.boot.version>
        <!--        1.1.0.M1 has the same problem-->
        <spring.content.version>1.0.0.M10</spring.content.version>
    </properties>

    <dependencyManagement>
        <dependencies>
            <dependency>
                <groupId>org.springframework.boot</groupId>
                <artifactId>spring-boot-dependencies</artifactId>
                <version>${spring.boot.version}</version>
                <type>pom</type>
                <scope>import</scope>
            </dependency>
        </dependencies>
    </dependencyManagement>

    <dependencies>
        <dependency>
            <groupId>com.github.paulcwarren</groupId>
            <artifactId>content-rest-spring-boot-starter</artifactId>
            <version>${spring.content.version}</version>
        </dependency>
        <dependency>
            <groupId>com.github.paulcwarren</groupId>
            <artifactId>content-s3-spring-boot-starter</artifactId>
            <version>${spring.content.version}</version>
        </dependency>
        <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-data-mongodb</artifactId>
        </dependency>
        <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-data-rest</artifactId>
        </dependency>
        <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-actuator</artifactId>
        </dependency>
        <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-configuration-processor</artifactId>
            <optional>true</optional>
        </dependency>
        <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-test</artifactId>
            <scope>test</scope>
            <exclusions>
                <exclusion>
                    <groupId>org.junit.vintage</groupId>
                    <artifactId>junit-vintage-engine</artifactId>
                </exclusion>
            </exclusions>
        </dependency>
    </dependencies>

    <build>
        <plugins>
            <plugin>
                <groupId>org.springframework.boot</groupId>
                <artifactId>spring-boot-maven-plugin</artifactId>
                <version>${spring.boot.version}</version>
                <executions>
                    <execution>
                        <goals>
                            <goal>repackage</goal>
                        </goals>
                    </execution>
                </executions>
            </plugin>
        </plugins>
    </build>
```

any idea whats causing this? thanks

**Comment by vachillo** [comment]
it seems the `Renderable<Document>` interface caused this error to be thrown. After removing it, no error but the store is still not exported as a resource.

**Comment by paulcwarren** [comment]
Thanks for this report @vachillo.  I will investigate and see if I can reproduce this for you/advise on troubleshooting to get to the root cause of this.

**Comment by paulcwarren** [comment]
I am unable to reproduce this.  spring-content is still compiled with Java 8 and I notice you are using Java 11.  Maybe that is relevant?  

It should be fine though and in fact I was able to put together [what I think you are doing](https://github.com/paulcwarren/issue-252) with a test and compile and run that test with Java 11 against Spring Boot 2.3.1 and Spring Content 1.1.0.M1.  I am using openjdk 11.  What are you using?  Can you git clone that repo and see if it works for you? 

A few additional comments:-
- something a little odd is going on as it seems to want to instantiate StoreFragments as a regular bean, rather than using the bean factory that Spring Content provides.  So the other possibility here maybe that Spring Boot auto-configuration isn't enabling the spring content s3 dependency.  Are you explicitly `@EnableS3Stores`?  Or letting auto-configuration do it?

- Just FYI; Spring Content 1.0.0 wont work with Spring Boot 2.3.  Spring Content 1.0.0 is designed to work with Spring Boot 2.2.

HTH

**Comment by paulcwarren** [comment]
Ping.  Any comment on this @vachillo ?

**Comment by vachillo** [comment]
sorry, had some other work get prioritized. That project did work and i was able to upload/download content. 
- I have `@EnableS3Stores` on my `@Configuration` file
- Good to know, ill stick with 1.1.0

I guess im misunderstanding how the repositories and content stores are interacting, and how that works with my use case. Basically i have some data repository that stores some document metadata, and the content store should store different binary files associated with that document. The two would be mostly unrelated, outside of the `ContentId` which needs to be resolved through path parameters to that document. the content should exposed as a resource of that document, something like `document/{id}/content/{id}`, where operations on `/document` would return metadata and `/content` returns binaries.

Also:
Trying to connect this with spring data elasticsearch repositories as well, but i'm getting an error on startup:
```
java.io.FileNotFoundException: class path resource [internal/org/springframework/content/elasticsearch/ElasticsearchConfig.class] cannot be opened because it does not exist
	at org.springframework.core.io.ClassPathResource.getInputStream(ClassPathResource.java:180) ~[spring-core-5.2.7.RELEASE.jar:5.2.7.RELEASE]
	at org.springframework.core.type.classreading.SimpleMetadataReader.getClassReader(SimpleMetadataReader.java:55) ~[spring-core-5.2.7.RELEASE.jar:5.2.7.RELEASE]
	at org.springframework.core.type.classreading.SimpleMetadataReader.<init>(SimpleMetadataReader.java:49) ~[spring-core-5.2.7.RELEASE.jar:5.2.7.RELEASE]
	at org.springframework.core.type.classreading.SimpleMetadataReaderFactory.getMetadataReader(SimpleMetadataReaderFactory.java:103) ~[spring-core-5.2.7.RELEASE.jar:5.2.7.RELEASE]
	at org.springframework.boot.type.classreading.ConcurrentReferenceCachingMetadataReaderFactory.createMetadataReader(ConcurrentReferenceCachingMetadataReaderFactory.java:86) ~[spring-boot-2.3.1.RELEASE.jar:2.3.1.RELEASE]
	at org.springframework.boot.type.classreading.ConcurrentReferenceCachingMetadataReaderFactory.getMetadataReader(ConcurrentReferenceCachingMetadataReaderFactory.java:73) ~[spring-boot-2.3.1.RELEASE.jar:2.3.1.RELEASE]
	at org.springframework.core.type.classreading.SimpleMetadataReaderFactory.getMetadataReader(SimpleMetadataReaderFactory.java:81) ~[spring-core-5.2.7.RELEASE.jar:5.2.7.RELEASE]
	at org.springframework.context.annotation.ConfigurationClassParser.asSourceClass(ConfigurationClassParser.java:695) ~[spring-context-5.2.6.RELEASE.jar:5.2.6.RELEASE]
	at org.springframework.context.annotation.ConfigurationClassParser$SourceClass.getRelated(ConfigurationClassParser.java:1089) ~[spring-context-5.2.6.RELEASE.jar:5.2.6.RELEASE]
	at org.springframework.context.annotation.ConfigurationClassParser$SourceClass.getAnnotationAttributes(ConfigurationClassParser.java:1070) ~[spring-context-5.2.6.RELEASE.jar:5.2.6.RELEASE]
	at org.springframework.context.annotation.ConfigurationClassParser.collectImports(ConfigurationClassParser.java:548) ~[spring-context-5.2.6.RELEASE.jar:5.2.6.RELEASE]
	at org.springframework.context.annotation.ConfigurationClassParser.getImports(ConfigurationClassParser.java:521) ~[spring-context-5.2.6.RELEASE.jar:5.2.6.RELEASE]
	at org.springframework.context.annotation.ConfigurationClassParser.doProcessConfigurationClass(ConfigurationClassParser.java:310) ~[spring-context-5.2.6.RELEASE.jar:5.2.6.RELEASE]
	at org.springframework.context.annotation.ConfigurationClassParser.processConfigurationClass(ConfigurationClassParser.java:249) ~[spring-context-5.2.6.RELEASE.jar:5.2.6.RELEASE]
	at org.springframework.context.annotation.ConfigurationClassParser.processImports(ConfigurationClassParser.java:599) ~[spring-context-5.2.6.RELEASE.jar:5.2.6.RELEASE]
```
I'm not trying to use the full text searching that content provides, just the repository.

**Comment by paulcwarren** [comment]
No problem.  I can appreciate your workload so thank you for getting back to me. 

Sounds like we are tracking 3 issues here:
- ConversionFailedException during startup
- Spring Content REST clarifications 
- An issue using spring content with spring data elasticsearch

I'lll try and address each one so we can keep track.

## ConversionFailedException
Unclear to me if you are still seeing this after comparing your project against my example project.  Please advise on this.

## Spring Content REST clarifications 
Its not clear to me if you are seeing an issue here, or not.  Or just seeking understanding.  So, perhaps I'll try and give a few pointers.  
- Your understanding sounds basically right.  
- Important to realize that out of the box Spring Content exports its Store into the same URI space as Spring Data REST and therefore `content-type` and `accept` headers are important if you want to get metadata responses from spring data when you expect them and binary responses from Spring Content when you expect them.  Also true if you want to use the renditions modules too to get renditions of content.
- Alternatives are to set the "path" attribute on the @StoreRestResource annotation, or set the `spring.content.rest.base-uri` property to specify a base uri for Spring Content REST resources.  base-uri is recommended.

Please let me know if you are seeing errors here.

## An issue using spring content with spring data elasticsearch
I believe this was fixed with [this commit](https://github.com/paulcwarren/spring-content/commit/0ff18784372fc0f883f8b97bdc37af1d347416de) but this isn't released yet so I'll cut 1.1.0.M2 tomorrow for you.

**Comment by paulcwarren** [comment]
1.1.0.M2 released that, I believe, will fix you spring data elasticsearch autoconfiguration issue.

**Comment by vachillo** [comment]
Ok great, that all makes sense. Lets say i wanted to export the repositories under another resource, something like `/resource/{resource-id}/documents/{id}/content/{id}`. Would this involve custom controller classes? basically, depending on the top-level resource, id want a certain repository to be exposed (which could be the same or different)

thanks for all your help, ill bump the version and try out the configuration.

**Comment by paulcwarren** [comment]
Yes, this would need to be implemented with a custom controller as neither Spring Data REST nor Spring Content REST can handle this use case.  However, you can use your Repository and Store interfaces to make implementing the controller a little easier.

HTH

**Comment by vachillo** [comment]
It seems I am getting another error. I dont even have any content stores set up yet, but its saying that there is a missing class definition:
```
java.lang.ClassNotFoundException: javax.persistence.Id
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:583) ~[na:na]
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178) ~[na:na]
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521) ~[na:na]
	at internal.org.springframework.content.rest.utils.DomainObjectUtils.getId(DomainObjectUtils.java:11) ~[spring-content-rest-1.1.0.M2.jar:na]
	at internal.org.springframework.content.rest.links.ContentLinksResourceProcessor.process(ContentLinksResourceProcessor.java:75) ~[spring-content-rest-1.1.0.M2.jar:na]
	at internal.org.springframework.content.rest.links.ContentLinksResourceProcessor.process(ContentLinksResourceProcessor.java:47) ~[spring-content-rest-1.1.0.M2.jar:na]
	at org.springframework.hateoas.server.mvc.RepresentationModelProcessorInvoker$DefaultProcessorWrapper.invokeProcessor(RepresentationModelProcessorInvoker.java:244) ~[spring-hateoas-1.1.0.RELEASE.jar:1.1.0.RELEASE]
	at org.springframework.hateoas.server.mvc.RepresentationModelProcessorInvoker.invokeProcessorsFor(RepresentationModelProcessorInvoker.java:159) ~[spring-hateoas-1.1.0.RELEASE.jar:1.1.0.RELEASE]
	at org.springframework.hateoas.server.mvc.RepresentationModelProcessorInvoker.invokeProcessorsFor(RepresentationModelProcessorInvoker.java:142) ~[spring-hateoas-1.1.0.RELEASE.jar:1.1.0.RELEASE]
	at org.springframework.hateoas.server.mvc.RepresentationModelProcessorHandlerMethodReturnValueHandler.handleReturnValue(RepresentationModelProcessorHandlerMethodReturnValueHandler.java:126) ~[spring-hateoas-1.1.0.RELEASE.jar:1.1.0.RELEASE]
	at org.springframework.web.method.support.HandlerMethodReturnValueHandlerComposite.handleReturnValue(HandlerMethodReturnValueHandlerComposite.java:82) ~[spring-web-5.2.7.RELEASE.jar:5.2.7.RELEASE]
.....
```
Am i missing a dependency somewhere? im using 
```
    <dependencyManagement>
        <dependencies>
            <dependency>
                <groupId>org.springframework.boot</groupId>
                <artifactId>spring-boot-dependencies</artifactId>
                <version>2.3.1.RELEASE</version>
            </dependency>
        </dependencies>
    </dependencyManagement>
```
in my pom file

**Comment by paulcwarren** [comment]
I managed to reproduce this.

Basically, the ContentLinksResourceProcessor is attempting to add content links to an entity response.   For this bean to be loaded then you either have `spring-content-rest-boot-starter` as a dependency on your classpath or `@Import(HypermediaConfiguration)` on a configuration class somewhere *and* you don't have hibernate on your classpath.  This makes sense if you are using spring data elasticsearch rather than spring data jpa.  So, I think the code is making an unsafe assumption here.

To that end I pushed a fix.  If you want to use the latest snapshot to try it then you will need to add this repository to your POM:

```xml
<repository>
    <id>snapshots</id>
    <name>nexus</name>
    <url>https://oss.sonatype.org/content/repositories/snapshots</url>
    <snapshots>
        <enabled>true</enabled>
    </snapshots>
</repository>
```

and update the version of your spring content dependencies to `1.1.0.M3-SNAPSHOT`.

All that said if you are writing your own controllers and not using spring content's controllers then you might be able to get away with just removing the `spring-content-rest/spring-content-rest-boot-starter` and staying on 1.1.0.M2.
